From ccaead5f2999e5fd295f300a8199a1fb2a48396a Mon Sep 17 00:00:00 2001
From: qwang2 <quanyang.wang@windriver.com>
Date: Tue, 3 Jul 2018 09:43:52 +0800
Subject: [PATCH] nxp-imx6: add clock property to pd_display node

In order to enter/exit "mem" power state, i.mx6sx should enable/disable
several power domains to save power. When resume from "mem" power state,
every power domain should enable all clocks which subjects to this power
domain, then do operations at power domain register, disable all clocks.
If not so, some peripheral controller will malfunction from resume, for
example pxp-dma which is included in display powerdomain. So we add 7
clocks to display power domain to avoid it.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |    4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index d0de399..859b054 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -900,6 +900,10 @@
 						reg = <2>;
 						#power-domain-cells = <0>;
 						power-supply = <&reg_soc>;
+						clocks = <&clks IMX6SX_CLK_IPG>,
+							<&clks IMX6SX_CLK_PXP_AXI>, <&clks IMX6SX_CLK_DISPLAY_AXI>,
+							<&clks IMX6SX_CLK_LCDIF1_PIX>, <&clks IMX6SX_CLK_LCDIF_APB>,
+							<&clks IMX6SX_CLK_LCDIF2_PIX>, <&clks IMX6SX_CLK_CSI>;
 					};
 
 					pd_pci: power-domain@3 {
-- 
1.7.9.5

